assets/src/components/activities/adaptive/AdaptiveAuthoring.tsx

```diff
@@ -68,19 +72,31 @@ const Adaptive = (
     }
   }, [selectedPart]);
 
+  // this keeps a reference to the actual part data of the selected part id in local state
   useEffect(() => {
     if (selectedPartId) {
       const part = parts.find((p) => p.id === selectedPartId);
       if (part) {
-        setSelectedPart(part);
+        let capabilities = { ...defaultCapabilities };
+        // attempt to get an instance of the part class
+        const PartClass = customElements.get(part.type);
+        if (PartClass) {
+          const instance = new PartClass();
+          if (instance.getCapabilities) {
```

Comment:
![image](https://user-images.githubusercontent.com/633004/133865900-a29ae389-44c6-4e40-bd85-8ce2ffbf5f65.png)

Reply:
hmm it didn't complain for me, nor did github...

Reply:
yeah, i didn't see any compilation errors and all the tests passed... so 🤷‍♂️